src/createServer.js

```diff
@@ -2,9 +1,37 @@
+const http = require('http');
+const { sendTextResponse } = require('./sendResponse');
+const { readFileByPath } = require('./getFilePath');
+const { getError } = require('./error');
+const url = require('url');
 
-function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
-}
+const createServer = () => {
+  const server = http.createServer(async (req, res) => {
+    try {
+      const normalizedUrl = new url.URL(
+        req.url || '',
+        `http://${req.headers.host}`,
+      );
 
-module.exports = {
-  createServer,
+      const requestedPath = normalizedUrl.pathname.slice(1);
+      const error = getError(requestedPath);
+
+      if (error) {
+        return sendTextResponse(res, ...error);
+      }
+
+      try {
+        const { data } = await readFileByPath(requestedPath);
+
+        return sendTextResponse(res, 200, data, 'OK');
+      } catch (err) {
+        sendTextResponse(res, 404, err.message);
```

Comment:
The error message for a 404 response should be more descriptive. Consider updating it to something like 'File not found' to provide clearer feedback to the user.